target/linux/ramips/image/mt7621.mk

```diff
@@ -2504,14 +2480,15 @@ define Device/ubnt_edgerouter_common
   $(Device/dsa-migration)
   $(Device/uimage-lzma-loader)
   DEVICE_VENDOR := Ubiquiti
-  IMAGE_SIZE := 256768k
+  IMAGE_SIZE := 259840k
   FILESYSTEMS := squashfs
-  KERNEL_SIZE := 3145728
-  KERNEL_INITRAMFS := $$(KERNEL) | \
-	ubnt-erx-factory-image $(KDIR)/tmp/$$(KERNEL_INITRAMFS_PREFIX)-factory.tar
+  KERNEL_SIZE := 6144k
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
   DEVICE_PACKAGES += -wpad-basic-mbedtls -uboot-envtools
-  DEFAULT := n
+  DEVICE_COMPAT_VERSION := 2.0
+  DEVICE_COMPAT_MESSAGE :=  Partition table has been changed due to kernel size restrictions. \
+    Refer to the wiki page for instructions to migrate to the new layout: \
```

Comment:
What is the plan here? I couldn't find any instructions, but I might be blind :)

Reply:
@ynezz I will add instructions on wiki page, as soon as its merged into main (or just prior). Didnt want to confuse people with documentation for the new layout when it wasn't available yet.

Reply:
> What is the plan here? I couldn't find any instructions

I added some initial notes to the wiki. Of course until this PR is merged there are not actually any images available with the new kernel layout and hence also no actual images for the migration scripts to download...

https://openwrt.org/toh/ubiquiti/edgerouter_x_er-x_ka#migrating_to_linux_66_for_24xx

Reply:
What is the expected user experience? Sysupgrade without preserve settings or sysupgrade a factory image?
It may be better to include this in the compat message and the wiki can be additional information. If the wiki became unavailable this is self contained. 
Other compat messages include this basic level of information and I think it is a good precedent to follow. 

My 2cents.

Reply:
A direct sysupgrade is not possible as it will truncate the kernel at 3MB and brick the router. Factory images are also very constrained on this device, it is no longer possible to build a working factory image since a very long time. Previously there have been seperate community build factory images with minimal configs, but even that wont be possible with 24.xx and Linux 6.6. 

This leaves two options to upgrade

1. Run migration script as proposed in #15197 for backport to 23.05.x, this takes care of flashing the new kernel across both existing kernel partitions and migrate system without preserving settings.
2. Reinstall a fresh install from serial console and TFTP to boot initramfs.

Documenting the process clearly at this stage is hard until I know exactly what is happening with both of these PR's. Hence the somewhat vague compat message. I will happily circle back around and clarify documentation in the compat message (and wiki), once i have a clear idea of what will or what might not be merged.

Reply:
A direct sysupgrade is not possible as it will truncate the kernel at 3MB and brick the router. Factory images are also very constrained on this device, it is no longer possible to build a working factory image since a very long time. Previously there have been seperate community build factory images with minimal configs, but even that wont be possible with 24.xx and Linux 6.6.

This leaves two options to upgrade

* Run migration script as proposed in #15197  for backport to 23.05.x, this takes care of flashing the new kernel across both existing kernel partitions and migrate system without preserving settings.

* Reinstall a fresh install from serial console and TFTP to boot initramfs.

Documenting the process clearly at this stage is hard until I know exactly what is happening with both of these PR's. Hence the somewhat vague compat message. I will happily circle back around and clarify documentation in the compat message (and wiki), once i have a clear idea of what will or what might not be merged.